src/Apps/CollectorProfile/Routes/Saves2/Components/SelectListsForArtworkModal/useUpdateCollectionsForArtwork.ts

```diff
@@ -29,5 +50,69 @@ export const useUpdateCollectionsForArtwork = () => {
         }
       }
     `,
+    updater: (store, data) => {
+      const artwork = store.get(artworkID)
+
+      if (!artwork) {
+        return
+      }
+
+      const response = data.artworksCollectionsBatchUpdate?.responseOrError
+      const addedCounts = getCountsByLists(response?.addedToCollections)
+      const removedCounts = getCountsByLists(response?.removedFromCollections)
+
+      // Set `isSaved` field to `true` if artwork was saved in "All Saves"
+      if (addedCounts.default > 0) {
+        artwork.setValue(true, "isSaved")
+      }
+
+      // Set `isSaved` field to `false` if artwork was unsaved from "All Saves"
+      if (removedCounts.default > 0) {
+        artwork.setValue(false, "isSaved")
+      }
+
+      const entity = artwork.getLinkedRecord("collectionsConnection", {
+        first: 0,
+        default: false,
+        saves: true,
+      })
+
+      if (!entity) {
+        return
+      }
+
+      /**
+       * Update `totalCount` field, based on which we decide
+       * whether to display the manage lists for artwork modal or
+       * immediately remove artwork from "All Saves"
+       */
+      const prevValue = (entity.getValue("totalCount") ?? 0) as number
+      const newValue = prevValue + addedCounts.custom - removedCounts.custom
+
+      entity.setValue(newValue, "totalCount")
+    },
   })
 }
+
+const getCountsByLists = (lists: ListEntity) => {
+  const listEntities = lists ?? []
+  const defaultCounts: Counts = {
+    custom: 0,
+    default: 0,
+  }
+
+  return listEntities.reduce((acc, list) => {
+    if (!list) {
+      return acc
+    }
+
+    const isDefault = list.name === "Saved Artwork"
```

Comment:
TODO Mention people
do we have the ability to return predefined `internalId` for "All Saves" list? For example, `all-saves'

Reply:
or is it better to do check in this way (if we can't add the changes 👆)

```javascript
// src/Apps/CollectorProfile/Routes/Saves2/Components/SelectListsForArtworkModal/SelectListsForArtworkModal.tsx
const savedCollection = me?.defaultSaves
const collections = savedCollection ? [savedCollection, ...nodes] : nodes
const { submitMutation } = useUpdateCollectionsForArtwork({
  artworkID: state.artwork!.id,
  allSavesListId: me?.defaultSaves?.internalID // 👈
})

// src/Apps/CollectorProfile/Routes/Saves2/Components/SelectListsForArtworkModal/useUpdateCollectionsForArtwork.ts
const addedCounts = getCountsByLists(response?.addedToCollections, allSavesListId)
const removedCounts = getCountsByLists(response?.removedFromCollections, allSavesListId)
...
const getCountsByLists = (lists: ListEntity) => {
  ...

  return listEntities.reduce((acc, list) => {
    ...
    const isDefault = list.internalID === allSavesListId
    ...
  }, defaultCounts)
}
```

Reply:
Can we use the default+saves boolean attributes instead of matching on the list name?

The default "All Saves" (currently "Saved Artwork") collection should consistently have `default=true` and `saves=false`.

```graphql
{
  me {
    collectionsConnection(first: 1, default: true, saves: false) {
      edges {
        node {
          name
          default
          saves
        }
      }
    }
  }
}
```

Reply:
totally forgot about these fields

Reply:
Addressed here: https://github.com/artsy/force/pull/12041/commits/e364c42066e4f296bc5a922faa247381f5ee6399